web/src/components/apps/NodeRow.jsx

```diff
@@ -1,62 +1,57 @@
 import React from "react";
 import classNames from "classnames";
 import Loader from "../shared/Loader";
-import ReactTooltip from "react-tooltip";
 import { rbacRoles } from "../../constants/rbac";
 import { getPercentageStatus, Utilities } from "../../utilities/utilities";
 
 export default function NodeRow(props) {
   const { node } = props;
 
-  let drainDeleteNode;
-  if (
-    props.drainNode &&
-    Utilities.sessionRolesHasOneOf([rbacRoles.CLUSTER_ADMIN])
-  ) {
-    if (
-      !props.drainNodeSuccessful &&
-      props.drainingNodeName &&
-      props.drainingNodeName === node?.name
-    ) {
-      drainDeleteNode = (
-        <div className="flex flex-auto alignItems--center">
-          <span className="u-marginRight--5">
-            <Loader size="25" />
-          </span>
-          <span className="u-fontSize--normal u-textColor--secondary u-fontWeight--medium">
-            Draining Node
-          </span>
-        </div>
-      );
-    } else if (
-      props.drainNodeSuccessful &&
-      props.drainingNodeName === node?.name
-    ) {
-      drainDeleteNode = (
-        <div className="flex flex-auto alignItems--center">
-          <span className="u-marginRight--5 icon checkmark-icon" />
-          <span className="u-fontSize--normal u-textColor--secondary u-fontWeight--medium">
-            Node successfully drained
-          </span>
-        </div>
-      );
-    } else {
-      drainDeleteNode = (
-        <div className="flex-auto flex-column justifyContent--center">
-          <button
-            onClick={() =>
-              node?.canDelete
-                ? props.deleteNode(node?.name)
-                : props.drainNode(node?.name)
-            }
-            className="btn secondary red"
-          >
-            {node?.canDelete ? "Delete node" : "Drain node"}
-          </button>
-        </div>
-      );
+  const DrainDeleteNode = () => {
```

Comment:
I considered refactoring this file, extracting some into components but felt like it was unnecessary. What do you think?

Reply:
I think this is fine. I haven't used styled components so maybe later you can show me how it could make this easier to read.